### PR TITLE
Migrated notebook cell modifier to scope handler

### LINKED ISSUE #1053: Migrate `notebookCell` to use a scope handler

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1053

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

**Review by pokey — COMMENTED** [review]
I agree we should try a simple migration here before doing something fancy, but this feels pretty hacky to me.  Let's discuss

File: packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type { Direction } from "@cursorless/common";
+import { NestedScopeHandler } from ".";
+import { NotebookCellTarget } from "../../targets";
+import { TargetScope } from "./scope.types";
+
+export default class NotebookCellScopeHandler extends NestedScopeHandler {
+  public readonly iterationScopeType = { type: "token" } as const;
```
Comment (pokey) [inline packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts:7]: we should find a way to make "every cell" throw an exception

File: packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import type { Direction } from "@cursorless/common";
+import { NestedScopeHandler } from ".";
+import { NotebookCellTarget } from "../../targets";
+import { TargetScope } from "./scope.types";
+
+export default class NotebookCellScopeHandler extends NestedScopeHandler {
```
Comment (pokey) [inline packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts:6]: should not be a nested scope handler

Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/NotebookCellScopeHandler.ts]: fixed

**Comment by pokey** [comment]
Ah right forgot about that. There's code in our VSCode impl that deals with some of that fwiw

**Comment by AndreasArvidsson** [comment]
Yeah but in a scope handler things should be quite generic so this is quite a pickle

**Comment by AndreasArvidsson** [comment]
Nope this one was a bit devious. Definitely not a good first issue! :D

**Comment by AndreasArvidsson** [comment]
To be honest I don't really have the time to do a lot of notebook stuffs at the moment. I think we should either just merge the simpler version I had initially, we can still throw an error on the iteration scope. Or just close this for now.

**Comment by pokey** [comment]
Feel free to assign it to me and I'll take a look when I get a minute

**Comment by AndreasArvidsson** [comment]
I don't think that this is a priority at the moment

**Pull request closed without merging**